pymc/tests/test_distributions.py

```diff
@@ -1769,6 +1769,30 @@ def test_poisson(self):
             {"mu": Rplus},
         )
 
+    # note we're only checking this distribution for lambda=0 where it's equivalent to Poisson.
+    # if you know of a better test case, please add it!
+    def test_generalized_poisson(self):
+        self.check_logp(
+            GeneralizedPoisson,
+            Nat,
+            {"theta": Rplus, "lam": Domain([0.0, 0.0, 0.0])},
+            lambda value, mu: sp.poisson.logpmf(value, mu),
+            skip_paramdomain_outside_edge_test=True,
+        )
+        self.check_logcdf(
+            GeneralizedPoisson,
+            Nat,
+            {"theta": Rplus, "lam": Domain([0.0, 0.0, 0.0])},
+            lambda value, mu: sp.poisson.logcdf(value, mu),
+            skip_paramdomain_outside_edge_test=True,
+        )
+        self.check_selfconsistency_discrete_logcdf(
+            GeneralizedPoisson,
+            Nat,
+            {"theta": Rplus, "lam": Domain([0.0, 0.0, 0.0])},
```

Comment:
```suggestion
            {"theta": Rplus, "lam": Domain([0.0, 0.0, 0.0])},
```
We can test different lam here. This is a good way to test our logp/logcdf are at least consistent in the cases where the distribution is not a Poisson